### ISSUE TITLE
TeaVM FreeType is not working with LibGDX 1.3.0 release

### ISSUE THREAD

**Issue by marcphilippebeaujean-abertay** [xpenatan/gdx-teavm#138]
Hey there,

I updated to a new version of libgdx dependencies and got this error while compiling in TeaVM:
| Class: com/badlogic/gdx/graphics/g2d/freetype/FreeTypeFontGenerator.java:533
| Method: com.badlogic.gdx.graphics.g2d.freetype.FreeTypeFontGenerator.createGlyph(CLcom/badlogic/gdx/graphics/g2d/freetype/FreeTypeFontGenerator$FreeTypeBitmapFontData;Lcom/badlogic/gdx/graphics/g2d/freetype/FreeTypeFontGenerator$FreeTypeFontParameter;Lcom/badlogic/gdx/graphics/g2d/freetype/FreeType$Stroker;FLcom/badlogic/gdx/graphics/g2d/PixmapPacker;)Lcom/badlogic/gdx/graphics/g2d/BitmapFont$Glyph;
| Text: Method com.badlogic.gdx.graphics.Pixmap.<init>(Lcom/badlogic/gdx/graphics/g2d/Gdx2DPixmap;)V

The dependencies were:

gdxTeaVMVersion=-SNAPSHOT
gdxVersion=1.12.1 -> 1.3.0
teaVMVersion=0.10.0

**Comment by xpenatan** [comment]
I'm waiting for 1.13.1 to update teavm backend.

**Comment by McMurat** [comment]
Is this fixed now? LibGDX 1.13.1 is released.

**Comment by McMurat** [comment]
Font generation fails for me with LibGDX 1.12.1. and the current master branch, too.

**Comment by xpenatan** [comment]
> Font generation fails for me with LibGDX 1.12.1. and the current master branch, too. There are no compilation errors but the generated texture is just transparent.

I did an investigation on this and is related to #137.

The fix is complicated.


**LibGDX** uses Java's `ByteBuffers` for handling OpenGL (GL) operations and other tasks. One of these tasks involves `Pixmap`, where:

- **In Java**, the `Pixmap` operations are handled via C code through the Java Native Interface (JNI). Every call to methods like `setPixel` updates both the Java `ByteBuffer` and the associated C memory.

- **When translating Java to JavaScript with TeaVM**, `ByteBuffer` internally uses an `Int8Array`. In JavaScript, I use **Emscripten** to convert C code to WebAssembly (Wasm), with the option to expand memory. All memory operations in WebAssembly are managed through a `HeapU8` buffer, which the JavaScript `Pixmap` code uses for its operations.

- **The initial approach** was to link the `HeapU8` buffer directly to `ByteBuffer`, but this led to a "detached state" problem. This happens because when the `HeapU8` memory grows, it creates a new buffer, invalidating the previous one.

- **A workaround** was to copy the `HeapU8` data into `ByteBuffer`, but this caused issue #137 where some `Pixmap` methods failed to make changes. This was because updates were applied to the copied buffer, not the actual `HeapU8`.

- **To summarize**:
  - If I use a direct reference to `HeapU8`, I encounter the detached state error.
  - If I copy `HeapU8`, `Pixmap` updates affect only the copy, not the real buffer.

- **Current exploration**: I'm working on a wrapper class that would always reference the most current `HeapU8` buffer even as it grows, but I'm experiencing some odd behaviors that need further investigation.

- **Another potential solution** is to use a fixed-size WebAssembly memory to avoid these issues altogether.

**Comment by xpenatan** [comment]
Gdx 1.13.1 is now at master, try snapshot to see if its working.
